transformer/transformer.go

```diff
@@ -202,20 +220,43 @@ func extractPreloads(dom *amphtml.DOM) []*rpb.Metadata_Preload {
 	return preloads
 }
 
-// setBaseURL derives the absolute base URL, and sets it on c.BaseURL. The value
-// is derived using the <base> href in the DOM, if it exists. If the href is
-// relative, it is parsed in the context of the document URL.
-// This must run after DocumentURL is set on the context.
-func setBaseURL(c *transformers.Context) {
-	if n, ok := htmlnode.FindNode(c.DOM.HeadNode, atom.Base); ok {
-		if v, ok := htmlnode.GetAttributeVal(n, "", "href"); ok {
-			if u, err := c.DocumentURL.Parse(v); err == nil {
-				c.BaseURL = u
-				return
+// defaultMaxAgeSeconds is the max-age to apply when there is an inline
+// amp-script without an explicit max-age. This is 1 day, to parallel the
+// security precautions put in place around service workers:
+// https://dev.chromium.org/Home/chromium-security/security-faq/service-worker-security-faq#TOC-Do-Service-Workers-live-forever-
+const defaultMaxAgeSeconds int32 = 86400  // number of seconds in a day
+
+// maxMaxAgeSeconds is the max duration of an SXG, per
+// https://wicg.github.io/webpackage/draft-yasskin-http-origin-signed-responses.html#signature-validity.
+const maxMaxAgeSeconds int32 = 7*86400
+
+// computeMaxAgeSeconds returns the suggested max-age based on the presence of
+// any inline <amp-script> tags on the page; callers should min() the return
+// value against any other they constraints they have (e.g. the max allowed
+// duration of an SXG).
+func computeMaxAgeSeconds(dom *amphtml.DOM) int32 {
```

Comment:
nit, optional. I prefer not to use "compute" in f'n names, since that's what f'ns do so it's redundant. I'd suggest domMaxAgeSeconds.